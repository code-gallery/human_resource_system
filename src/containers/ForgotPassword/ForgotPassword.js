import React from 'react'
import Notifications, { notify } from 'react-notify-toast'
import { NOTIFICATION_TIMEOUT, ROUTE_URL, getApiUrl } from 'containers/constants'
import Form from 'components/Abstract/Form'
import httpFetch from 'utils/httpFetch'
import 'containers/Register/style.css'

class ForgotPassword extends Form {
  constructor(props) {
    super(props)
    this.state = {
      email: '',
      sending: false
    }
    this.genericError = 'An error occured. Please try again.'
    this.onSubmit = this.onSubmit.bind(this)
  }
  componentDidMount() {
    document.body.classList.add('login-page')
  }

  componentWillUnmount() {
    document.body.classList.remove('login-page')
  }

  onSubmit(event) {
    event.preventDefault()
    const { email } = this.state
    if (email !== '' && email.indexOf('@') !== -1 && email.indexOf('.') !== -1) {
      this.setState({ sending: true })
      httpFetch(getApiUrl('resetPasswordRequest'), {
        method: 'POST',
        body: JSON.stringify({
          email
        })
      })
        .then((data) => {
          if (data.status === 'success') {
            notify.show(data.message, 'success', NOTIFICATION_TIMEOUT)
          } else {
            const msg = data.message || this.genericError
            notify.show(msg, 'error', NOTIFICATION_TIMEOUT)
          }
          this.setState({ sending: false })
        })
        .catch(() => {
          notify.show(this.genericError, 'error', NOTIFICATION_TIMEOUT)
          this.setState({ sending: false })
        })
    } else {
      notify.show('Please enter an email', 'error', NOTIFICATION_TIMEOUT)
    }
  }

  render() {
    const { email, sending } = this.state

    return (
      <form className="Login-page">
        <Notifications />
        <section className="Login-block">
          <h1 className="Login-logo">
            {/* external link - not react route - leave <a> */}
            <a href={ROUTE_URL.publicSiteHome}><span>APPII</span></a>
          </h1>
          <p className="Login-title">Reset your password</p>
          <div className="form-group">
            <input
              name="email"
              type="text"
              placeholder="Email"
              value={email}
              onChange={this.onChangeInput}
            />
          </div>
          <div className="form-group">
            <button className="border-btn" onClick={this.onSubmit} disabled={sending}>
              {!sending ? 'Send' : 'Please wait'}
            </button>
          </div>
        </section>
      </form>
    )
  }
}

export default ForgotPassword
