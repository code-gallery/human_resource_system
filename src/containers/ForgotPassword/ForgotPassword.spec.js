import React from 'react'
import renderer from 'react-test-renderer'
import ForgotPassword from './ForgotPassword'

describe('<ForgotPassword />', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <ForgotPassword />
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
