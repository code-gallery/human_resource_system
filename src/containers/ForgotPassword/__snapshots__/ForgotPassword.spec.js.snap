// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ForgotPassword /> renders correctly 1`] = `
<form
  className="Login-page"
>
  <div
    id="notification-wrapper"
  />
  <section
    className="Login-block"
  >
    <h1
      className="Login-logo"
    >
      <a
        href="https://appii.io"
      >
        <span>
          APPII
        </span>
      </a>
    </h1>
    <p
      className="Login-title"
    >
      Reset your password
    </p>
    <div
      className="form-group"
    >
      <input
        name="email"
        onChange={[Function]}
        placeholder="Email"
        type="text"
        value=""
      />
    </div>
    <div
      className="form-group"
    >
      <button
        className="border-btn"
        disabled={false}
        onClick={[Function]}
      >
        Send
      </button>
    </div>
  </section>
</form>
`;
